componentes/Post.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Image, Badge, Box } from "@chakra-ui/react";
+import { IDataPost } from "../types/types";
+import Link from "next/link";
+
+interface dataType {
```

Comment:
tipos e classes começam com letras maiusculas.

Tbm vale a pena usar um nome mais especifico que data

```suggestion
interface PostProps {
```